workshops/git_101/02-clone-commit-push-merge.md

```diff
@@ -0,0 +1,161 @@
+```
+
+Now any changes you make are in a separate branch from main called test.
+
+## Commiting Code
+
+1. In the HowWeEnable directory create a new file and look at the status of it in git.
+
+    ```shell
+    touch mynewfile.txt
+    git status
+    ```
+
+    Output of git status:
+
+    ```shell
+    Untracked files:
+    (use "git add <file>..." to include in what will be committed)
+        mynewfile.txt
+    ```
+
+2. Now let's git add so the file is tracked by git going forward, and git commit so the new file is ready to be pushed to the remote repository.
+
+    ```shell
+    git add mynewfile.txt
+    git status
+    ```
+
+    Output of git status:
+
+    ```shell
+    Changes to be committed:
+    (use "git restore --staged <file>..." to unstage)
+            new file:   _config.yml
+    ```
+
+3. Finally lets commit the code:
+
+    ```shell
+    git commit -m 'adding newfile.txt'
+    ```
+
+    Output:
+
+    ```shell
+    [main 60cdc2b] adding newfile.txt
+     1 file changed, 0 insertions(+), 0 deletions(-)
+    create mode 100644 newfile.txt
+    ```
+
+## Pushing Commit to Repository Server
+
+Just like it sounds, `git push` pushes the commits to the central repository. A best practice is to specify the repo you are pushing to (usually origin) and the branch for this example we will use test. `Note:` Generally upstream repositories will not allow pushing directly to main, so you will always need to either fork the repository, or create a branch.
+
+```shell
+git push origin test
+```
+
+Output:
+
+```shell
+Enumerating objects: 1, done.
+Counting objects: 100% (1/1), done.
+Delta compression using up to 8 threads
+Compressing objects: 100% (1/1), done.
+Writing objects: 100% (1/1), 1018 bytes | 339.00 KiB/s, done.
+Total 9 (delta 5), reused 0 (delta 0), pack-reused 0
+remote: Resolving deltas: 100% (1/1), completed with 1 local objects.
+To github.com:redhat-adsa/HowWeEnable.git
+   c0db901..6eaaf6d  test -> test
+```
+
+## Merging Code
+
+If you are the owner and want to merge a feature branch into main you can do so via `git merge <branchname>` from the main branch, however this is not a best practice. Best practice is to perform a pull request (sometimes called merge request) from the central server and have someone review the changes before they are merged into main. Below are links to each major repository servers instructions to perform a pull request:
+
+[Azure Devops](https://docs.microsoft.com/en-us/azure/devops/repos/git/pull-requests?view=azure-devops)
+
+[Bitbucket](https://www.atlassian.com/git/tutorials/making-a-pull-request)
+
+[Github](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request)
+
+[Gitlab](https://docs.gitlab.com/ee/user/project/merge_requests/creating_merge_requests.html)
+
+Please create a pull request following above documentation to your repository you created above from the `test` branch into the `main` branch of your fork.
```

Comment:
Add a link to next steps or back to the project root?

Reply:
```suggestion
Please create a pull request following above documentation to your repository you created above from the `test` branch into the `main` branch of your fork.

---

[Next](workshops/git101/03-advanced-git.md)

```